### ISSUE TITLE
Pasta failed error when running container with a privileged podman in container setup 

### ISSUE THREAD

**Issue by brianmcarey** [containers/podman#22907]
### Issue Description

KubeVirt uses a privileged podman in container setup to carry out a lot of CI testing. The containers.conf and other configuration files for this podman in container setup can be found here - https://github.com/kubevirt/project-infra/tree/main/images/bootstrap

This podman in container setup worked with podman v4 however it started to fail after upgrading to Fedora40 / podman v5. 

After upgrading to podman v5 the following error occurs when trying to run containers
```
podman run hello-world
Resolved "hello-world" as an alias (/etc/containers/registries.conf.d/000-shortnames.conf)
Trying to pull quay.io/podman/hello:latest...
Getting image source signatures
Copying blob 81df7ff16254 done   |
Copying config 5dd467fce5 done   |
Writing manifest to image destination
Error: setting up Pasta: pasta failed with exit code 1:
Don't run as root. Changing to nobody...
```

### Steps to reproduce the issue

Steps to reproduce the issue
1. With rootful podman run the following - this image includes podman v5.0.3
    `podman run --privileged -e PODMAN_IN_CONTAINER_ENABLED=true quay.io/bcarey/bootstrap:v20240605-064d50c-amd64 podman run hello-world`

### Describe the results you received

The container fails to run.

### Describe the results you expected

The container runs successfully

For example with the same podman in container setup with podman v4.9.4:
```
podman run --privileged -e PODMAN_IN_CONTAINER_ENABLED=true quay.io/kubevirtci/bootstrap:v20240517-5a9a066 podman run hello-world
Enabling IPV6.
net.ipv6.conf.all.disable_ipv6 = 0
net.ipv6.conf.all.forwarding = 1
Podman in Container enabled, initializing in podman compatible mode...
Waiting for podman socket to be ready, sleeping for 1 seconds.
================================================================================
Done setting up podman in container.
fatal: not a git repository (or any of the parent directories): .git
+ podman run hello-world
Resolved "hello-world" as an alias (/etc/containers/registries.conf.d/000-shortnames.conf)
Trying to pull quay.io/podman/hello:latest...
Getting image source signatures
Copying blob sha256:81df7ff16254ed9756e27c8de9ceb02a9568228fccadbf080f41cc5eb5118a44
Copying config sha256:5dd467fce50b56951185da365b5feee75409968cbab5767b9b59e325fb2ecbc0
Writing manifest to image destination
!... Hello Podman World ...!

         .--"--.           
       / -     - \         
      / (O)   (O) \        
   ~~~| -=(,Y,)=- |         
    .---. /`  \   |~~      
 ~/  o  o \~~~~.----. ~~   
  | =(X)= |~  / (O (O) \   
   ~~~~~~~  ~| =(Y_)=-  |   
  ~~~~    ~~~|   U      |~~ 

Project:   https://github.com/containers/podman
Website:   https://podman.io
Desktop:   https://podman-desktop.io
Documents: https://docs.podman.io
YouTube:   https://youtube.com/@Podman
X/Twitter: @Podman_io
Mastodon:  @Podman_io@fosstodon.org
+ EXIT_VALUE=0
+ set +o xtrace
Cleaning up after podman in container.
================================================================================
Cleaning up after podman
8e277cd053339cd5e287471aeeae1484253f15ecf081121b1631e66cf499e823
/usr/local/bin/runner.sh: line 50: wait: pid 8 is not a child of this shell
================================================================================
Done cleaning up after podman in container.
```

### podman info output

```yaml
podman info
host:
  arch: amd64
  buildahVersion: 1.35.4
  cgroupControllers:
  - cpuset
  - cpu
  - io
  - memory
  - hugetlb
  - pids
  cgroupManager: cgroupfs
  cgroupVersion: v2
  conmon:
    package: conmon-2.1.10-1.fc40.x86_64
    path: /usr/bin/conmon
    version: 'conmon version 2.1.10, commit: '
  cpuUtilization:
    idlePercent: 99
    systemPercent: 0.18
    userPercent: 0.82
  cpus: 12
  databaseBackend: sqlite
  distribution:
    distribution: fedora
    variant: container
    version: "40"
  eventLogger: file
  freeLocks: 2048
  hostname: a80c393d31e6
  idMappings:
    gidmap: null
    uidmap: null
  kernel: 6.8.10-300.fc40.x86_64
  linkmode: dynamic
  logDriver: k8s-file
  memFree: 10303053824
  memTotal: 67162460160
  networkBackend: netavark
  networkBackendInfo:
    backend: netavark
    dns:
      package: aardvark-dns-1.10.0-1.fc40.x86_64
      path: /usr/libexec/podman/aardvark-dns
      version: aardvark-dns 1.10.0
    package: netavark-1.10.3-3.fc40.x86_64
    path: /usr/libexec/podman/netavark
    version: netavark 1.10.3
  ociRuntime:
    name: crun
    package: crun-1.15-1.fc40.x86_64
    path: /usr/bin/crun
    version: |-
      crun version 1.15
      commit: e6eacaf4034e84185fd8780ac9262bbf57082278
      rundir: /run/crun
      spec: 1.0.0
      +SYSTEMD +SELINUX +APPARMOR +CAP +SECCOMP +EBPF +CRIU +LIBKRUN +WASM:wasmedge +YAJL
  os: linux
  pasta:
    executable: /bin/pasta
    package: passt-0^20240510.g7288448-1.fc40.x86_64
    version: |
      pasta 0^20240510.g7288448-1.fc40.x86_64
      Copyright Red Hat
      GNU General Public License, version 2 or later
        <https://www.gnu.org/licenses/old-licenses/gpl-2.0.html>
      This is free software: you are free to change and redistribute it.
      There is NO WARRANTY, to the extent permitted by law.
  remoteSocket:
    exists: true
    path: /run/podman/podman.sock
  security:
    apparmorEnabled: false
    capabilities: CAP_CHOWN,CAP_DAC_OVERRIDE,CAP_FOWNER,CAP_FSETID,CAP_KILL,CAP_NET_BIND_SERVICE,CAP_SETFCAP,CAP_SETGID,CAP_SETPCAP,CAP_SETUID,CAP_SYS_CHROOT
    rootless: false
    seccompEnabled: true
    seccompProfilePath: /usr/share/containers/seccomp.json
    selinuxEnabled: false
  serviceIsRemote: false
  slirp4netns:
    executable: ""
    package: ""
    version: ""
  swapFree: 8588357632
  swapTotal: 8589930496
  uptime: 121h 13m 35.00s (Approximately 5.04 days)
  variant: ""
plugins:
  authorization: null
  log:
  - k8s-file
  - none
  - passthrough
  - journald
  network:
  - bridge
  - macvlan
  - ipvlan
  volume:
  - local
registries:
  search:
  - registry.fedoraproject.org
  - registry.access.redhat.com
  - docker.io
  - quay.io
store:
  configFile: /etc/containers/storage.conf
  containerStore:
    number: 0
    paused: 0
    running: 0
    stopped: 0
  graphDriverName: overlay
  graphOptions:
    overlay.imagestore: /var/lib/shared-images
    overlay.mountopt: nodev
  graphRoot: /var/lib/containers/storage
  graphRootAllocated: 510389125120
  graphRootUsed: 314424713216
  graphStatus:
    Backing Filesystem: btrfs
    Native Overlay Diff: "true"
    Supports d_type: "true"
    Supports shifting: "true"
    Supports volatile: "true"
    Using metacopy: "false"
  imageCopyTmpDir: /var/tmp
  imageStore:
    number: 0
  runRoot: /run/containers/storage
  transientStore: false
  volumePath: /var/lib/containers/storage/volumes
version:
  APIVersion: 5.0.3
  Built: 1715299200
  BuiltTime: Fri May 10 00:00:00 2024
  GitCommit: ""
  GoVersion: go1.22.2
  Os: linux
  OsArch: linux/amd64
  Version: 5.0.3
```

### Podman in a container

Yes

### Privileged Or Rootless

Privileged

### Upstream Latest Release

No

### Additional environment details

Additional environment details

### Additional information

Additional information like issue happens only occasionally or issue happens with a particular architecture or on a particular setting

**Comment by Luap99** [comment]
Your outer and inner container is started as root (uid 0), right?

**Comment by brianmcarey** [comment]
> Your outer and inner container is started as root (uid 0), right?

Yes that's correct.

**Comment by Luap99** [comment]
Please unset _CONTAINERS_USERNS_CONFIGURED from your image as workaround, see https://github.com/containers/podman/issues/22791

Or alternatively update your image to podman 5.1.1 it should contain the required bug fixes to make it work but honestly have no idea why _CONTAINERS_USERNS_CONFIGURED was ever added

**Comment by brianmcarey** [comment]
> Please unset _CONTAINERS_USERNS_CONFIGURED from your image as workaround, see #22791
> 
> Or alternatively update your image to podman 5.1.1 it should contain the required bug fixes to make it work but honestly have no idea why _CONTAINERS_USERNS_CONFIGURED was ever added

That looks to work!

<details>

```
podman run --privileged -e PODMAN_IN_CONTAINER_ENABLED=true quay.io/bcarey/bootstrap:v20240605-064d50c-amd64 /bin/sh -c "unset _CONTAINERS_USERNS_CONFIGURED && podman run hello-world"
Enabling IPV6.
net.ipv6.conf.all.disable_ipv6 = 0
net.ipv6.conf.all.forwarding = 1
Podman in Container enabled, initializing in podman compatible mode...
Waiting for podman socket to be ready, sleeping for 1 seconds.
================================================================================
Done setting up podman in container.
fatal: not a git repository (or any of the parent directories): .git
+ /bin/sh -c 'unset _CONTAINERS_USERNS_CONFIGURED && podman run hello-world'
Resolved "hello-world" as an alias (/etc/containers/registries.conf.d/000-shortnames.conf)
Trying to pull quay.io/podman/hello:latest...
Getting image source signatures
Copying blob sha256:81df7ff16254ed9756e27c8de9ceb02a9568228fccadbf080f41cc5eb5118a44
Copying config sha256:5dd467fce50b56951185da365b5feee75409968cbab5767b9b59e325fb2ecbc0
Writing manifest to image destination
!... Hello Podman World ...!

         .--"--.           
       / -     - \         
      / (O)   (O) \        
   ~~~| -=(,Y,)=- |         
    .---. /`  \   |~~      
 ~/  o  o \~~~~.----. ~~   
  | =(X)= |~  / (O (O) \   
   ~~~~~~~  ~| =(Y_)=-  |   
  ~~~~    ~~~|   U      |~~ 

Project:   https://github.com/containers/podman
Website:   https://podman.io
Desktop:   https://podman-desktop.io
Documents: https://docs.podman.io
YouTube:   https://youtube.com/@Podman
X/Twitter: @Podman_io
Mastodon:  @Podman_io@fosstodon.org
```

</details>

**Comment by Luap99** [comment]
You can drop the line here https://github.com/kubevirt/project-infra/blob/5fae7e4a6393bb9f06a33eae5ef32c3d8736213c/images/bootstrap/Dockerfile#L109

There is no harm in doing that and I really don't understand why this was ever added to our Containerfile in the first place.

But since the regression should be fixes in 5.1.1 I close this one.
